src/LocalFeedConfigs.php

```diff
@@ -0,0 +1,119 @@
+<?php //phpcs:disable WordPress.WP.AlternativeFunctions --- Uses FS read/write in order to reliable append to an existing file.
+/**
+ * Class responsible for managing local feed configurations.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @version     x.x.x
+ */
+namespace Automattic\WooCommerce\Pinterest;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Handling feed files generation.
+ *
+ * Singleton pattern used.
+ * Prevent application from having multiple instances of configuration.
+ * At the same time allow distributed use of configurations.
+ */
+class LocalFeedConfigs {
+
+	/**
+	 * Array of local feed configurations.
+	 *
+	 * @var array $feeds_configurations
+	 */
+	private $feeds_configurations = null;
```

Comment:
Since this variable is marked as type `array`, it should be instantiated as an empty array.

```suggestion
	private $feeds_configurations = [];
```

Reply:
👍